lib/service/connect.go

```diff
@@ -76,10 +76,17 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 		}
 		process.log.Errorf("%v failed to establish connection to cluster: %v.", role, err)
 
+		// Used for testing that auth service will attempt to reconnect in the provided duration.
+		select {
+		case process.connectFailureC <- retry.Duration():
+		default:
+		}
+
+		startedWait := process.Clock.Now()
 		// Wait in between attempts, but return if teleport is shutting down
-		process.log.Debugf("Retrying connection to auth server in %v", retry)
 		select {
-		case <-retry.After():
+		case t := <-retry.After():
```

Comment:
The time sent over `connectFailureC` and the time waited in `After` will be slightly different because of jitter, is it ok in practice?

Reply:
(same goes for lib/services/watcher.go line 201)

Reply:
Yeah this is unfortunately a limitation of `Retry` that I didn't want to try and change in this PR. The tests introduced in this PR all advance the fake clock by a multiple of the `Duration` sent over the channel to ensure that we move far enough in the future to trigger `After`.